File: solutions/php/eliuds-eggs/1/EliudsEggs.php
<?php
declare(strict_types=1);

class EliudsEggs
{
    function eggCount(int $displayValue): int
    {
        $count = 0;
        $n = $displayValue;
        while ($n > 0) {
            if (($n & 1) === 1) {
                $count += 1;
            }
            $n >>= 1;
        }
        return $count;
    }
}

File: solutions/lua/eliuds-eggs/1/eliuds-eggs.lua
local EliudsEggs = {}

function EliudsEggs.egg_count(number)
    local count = 0
    local n = number
    while n > 0 do
        if (n % 2) == 1 then
            count = count + 1
        end
        n = math.floor(n / 2)
    end
    return count
end

return EliudsEggs
